src/Storages/System/StorageSystemDDLWorkerQueue.cpp

```diff
@@ -0,0 +1,75 @@
+#include <algorithm>
+#include <string>
+
+#include "StorageSystemDDLWorkerQueue.h"
+
+#include <Columns/ColumnArray.h>
+#include <DataTypes/DataTypeArray.h>
+#include <DataTypes/DataTypeString.h>
+#include <Interpreters/Context.h>
+#include <Common/ZooKeeper/ZooKeeper.h>
+
+
+namespace DB
+{
+NamesAndTypesList StorageSystemDDLWorkerQueue::getNamesAndTypes()
+{
+    return {
+        {"name", std::make_shared<DataTypeString>()}, // query_<id>
+        {"active", std::make_shared<DataTypeArray>(std::make_shared<DataTypeString>())}, // list of host_fqdn
+        {"finished", std::make_shared<DataTypeArray>(std::make_shared<DataTypeString>())}, // list of host_fqdn
+    };
+}
+
+
+void StorageSystemDDLWorkerQueue::fillData(MutableColumns & res_columns, const Context & context, const SelectQueryInfo &) const
+{
+    zkutil::ZooKeeperPtr zookeeper = context.getZooKeeper();
+
+    String ddl_zookeeper_path = config.getString("distributed_ddl.path", "/clickhouse/task_queue/ddl/");
```

Comment:
Should use `distributed_ddl.path` config directive

Reply:
@azat not sure if I understand you completely. Are you implying if I need to a check similar to:`if ( config().has("distributed_ddl"))` before actually doing a `config.getString(...)` ?

Reply:
Egh, sorry, by some reason I missed that out already did this